Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeActionHandler.cpp

```diff
@@ -0,0 +1,132 @@
+        }
+    }
+
+    void ComponentModeActionHandler::OnActionContextModeRegistrationHook()
+    {
+        m_actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(
+            m_actionManagerInterface,
+            "ComponentModeActionHandler - could not get ActionManagerInterface on ComponentModeActionHandler "
+            "OnActionUpdaterRegistrationHook.");
+
+        m_actionManagerInterface->RegisterActionContextMode(EditorMainWindowActionContextIdentifier, "AnyComponentMode");
+    }
+
+    void ComponentModeActionHandler::OnActionUpdaterRegistrationHook()
+    {
+        m_actionManagerInterface->RegisterActionUpdater(ComponentModeChangedUpdaterIdentifier);
+    }
+
+    void ComponentModeActionHandler::OnActionRegistrationHook()
+    {
+        m_hotKeyManagerInterface = AZ::Interface<AzToolsFramework::HotKeyManagerInterface>::Get();
+        AZ_Assert(
+            m_hotKeyManagerInterface,
+            "ComponentModeActionHandler - could not get HotKeyManagerInterface on ComponentModeActionHandler OnActionRegistrationHook.");
+
+        // Add default actions for every Component Mode
+        
+        // Exit Component Mode
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.componentMode.end";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = "Done";
+            actionProperties.m_description = "Return to normal viewport editing";
+            actionProperties.m_category = "Component Modes";
+
+            m_actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    ComponentModeFramework::ComponentModeSystemRequestBus::Broadcast(
+                        &ComponentModeFramework::ComponentModeSystemRequests::EndComponentMode);
+                }
+            );
+
+            // Only add this to the component modes, not default.
+            m_actionManagerInterface->AssignModeToAction("SomeMode", actionIdentifier);
+
+            m_hotKeyManagerInterface->SetActionHotKey(actionIdentifier, "Esc");
+        }
+    }
+
+    void ComponentModeActionHandler::OnMenuBindingHook()
+    {
+        m_menuManagerInterface = AZ::Interface<AzToolsFramework::MenuManagerInterface>::Get();
+        AZ_Assert(
+            m_menuManagerInterface,
+            "ComponentModeActionHandler - could not get MenuManagerInterface on ComponentModeActionHandler OnMenuBindingHook.");
+
+        m_menuManagerInterface->AddSeparatorToMenu(EditMenuIdentifier, 10000);
+        m_menuManagerInterface->AddActionToMenu(EditMenuIdentifier, "o3de.action.componentMode.end", 10001);
+    }
+
+    void ComponentModeActionHandler::ComponentModeStarted([[maybe_unused]] const AZ::Uuid& componentType)
```

Comment:
update: I believe this can be replaced with... `OnEditorModeActivated` on the `ViewportEditorModeNotificationsBus`

Reply:
Thanks for updating